demo/index.html

```diff
@@ -20,7 +23,62 @@
       <h3>Basic mtz-ajax-interceptor demo</h3>
       <demo-snippet>
         <template>
-          <mtz-ajax-interceptor></mtz-ajax-interceptor>
+          <test-el></test-el>
+          <dom-module id="test-el">
+            <template>
+              <mtz-ajax-interceptor>
+                <mtz-status-code-interceptor
+                  status="200"
+                  on-intercept="statusCodeInterceptor"
+                ></mtz-status-code-interceptor>
+                <template is="dom-if" if="[[headerInterceptor]]" restamp>
+                  <mtz-status-code-interceptor
+                    status="200"
+                    on-intercept="statusCodeInterceptor"
+                  ></mtz-status-code-interceptor>
+                  <mtz-presend-header-interceptor
+                    key="header"
+                    value="test"
+                  ></mtz-presend-header-interceptor>
+                </template>
+
+                <iron-ajax url="/fake" id="ajax" bubbles></iron-ajax>
+              </mtz-ajax-interceptor>
+              <p>
+                [[statusCodeIntercepted]]
+              </p>
+              <label>
+                Inject additional interceptors
+                <input type="checkbox" checked="{{headerInterceptor::change}}">
+              </label>
+              <br>
+              <button on-tap="intercept">Trigger Interceptors</button>
+            </template>
+            <script>
+              Polymer({
+                is: 'test-el',
+                properties: {
+                  headerInterceptor: {
+                    type: Boolean,
+                    value: false
+                  },
+                  statusCodeIntercepted: {
+                    type: String,
+                    value: ''
+                  }
+                },
+                listeners: {
+                  intercept: '_log'
+                },
+                intercept() {
+                  this.$.ajax.generateRequest();
+                },
+                statusCodeInterceptor() {
+                  this.statusCodeIntercepted = this.statusCodeIntercepted.concat('intercepted! ');
```

Comment:
This is a visual representation going into the DOM via a binding, was easier to read if there was a space between each entry added since spamming it would end up with "intercepted! intercepted! ..."